File: frontend/src/dto/response/PaymentResponse.js
// dto/PaymentResponse.js


class PaymentResponse {


    constructor(data = {}) {


        this.id =
            data.id || 0;


        this.loanType =
            data.loanType || "";


        this.loanName =
            data.loanName || "";


        this.lenderName =
            data.lenderName || "";


        this.maskedAccountNumber =
            data.maskedAccountNumber || "";


        this.currentBalance =
            data.currentBalance || 0;


        this.paymentAmount =
            data.paymentAmount || 0;


        this.paymentFrequency =
            data.paymentFrequency || "";


        this.paymentDate =
            data.paymentDate || "";


        this.paymentMethod =
            data.paymentMethod || "";


        this.status =
            data.status || "Active";


        this.createdDate =
            data.createdDate || "";


    }

}


export default PaymentResponse;

File: backend/AppPortal.Api/ResponseModels/PaymentResponse.cs
namespace AppPortal.Api.DTOs;

public class PaymentResponse
{
    public int Id { get; set; }

    public int? LoanAccountId { get; set; }

    public string? LoanAccountNumber {get;set;}

    public string? PaymentDescription { get; set; }

    public decimal? PaymentAmount { get; set; }

    public DateTime? PaymentDate { get; set; }

    public string? Status { get; set; }

    public string? ConfirmationNumber { get; set; }

    public DateTime? CreatedDate { get; set; }

    public DateTime? CompletedDate { get; set; }


    // Loan Account Information

    public string? LoanType { get; set; }

    public string? LoanName { get; set; }

    public string? LenderName { get; set; }

    public string? MaskedAccountNumber { get; set; }

    public decimal? CurrentBalance { get; set; }

    public string? PaymentFrequency { get; set; }
}